test/extended/builds/pipeline.go

```diff
@@ -203,930 +333,669 @@ var _ = g.Describe("[Feature:Builds][Slow] openshift pipeline build", func() {
-			_, err = exutil.GetEndpointAddress(oc, "nodejs-mongodb-example")
-			o.Expect(err).NotTo(o.HaveOccurred())
+			g.By("clean up openshift resources for next potential run")
+			err = oc.Run("delete").Args("bc", "sample-pipeline-withenvs").Execute()
 		})
-	})
 
-	g.Context("Deleted pipeline strategy buildconfigs", func() {
-		g.BeforeEach(func() {
+		g.It("using the ephemeral template", func() {
+			defer cleanup()
 			setupJenkins(jenkinsEphemeralTemplatePath)
 
-			if os.Getenv(jenkins.EnableJenkinsMemoryStats) != "" {
-				ticker = jenkins.StartJenkinsMemoryTracking(oc, oc.Namespace())
-			}
+			g.By("Pipelines with maven slave")
 
-			g.By("waiting for builder service account")
-			err := exutil.WaitForBuilderAccount(oc.KubeClient().Core().ServiceAccounts(oc.Namespace()))
-			o.Expect(err).NotTo(o.HaveOccurred())
-		})
+			g.By("should build a maven project and complete successfully", func() {
 
-		g.AfterEach(func() {
-			if g.CurrentGinkgoTestDescription().Failed {
-				exutil.DumpPodStates(oc)
-				exutil.DumpPodLogsStartingWith("", oc)
-				exutil.DumpPersistentVolumeInfo(oc)
-			}
-			if os.Getenv(jenkins.EnableJenkinsMemoryStats) != "" {
-				ticker.Stop()
-			}
-		})
+				// instantiate the template
+				g.By(fmt.Sprintf("calling oc new-app -f %q", mavenSlavePipelinePath))
+				err := oc.Run("new-app").Args("-f", mavenSlavePipelinePath).Execute()
+				o.Expect(err).NotTo(o.HaveOccurred())
 
-		g.It("should not be recreated by the sync plugin", func() {
-			// create the bc
-			g.By(fmt.Sprintf("calling oc new-app -f %q", origPipelinePath))
-			err := oc.Run("new-app").Args("-f", origPipelinePath).Execute()
-			o.Expect(err).NotTo(o.HaveOccurred())
+				// start the build
+				g.By("starting the pipeline build and waiting for it to complete")
+				br, err := exutil.StartBuildAndWait(oc, "openshift-jee-sample")
+				if err != nil || !br.BuildSuccess {
+					exutil.DumpBuilds(oc)
+					exutil.DumpPodLogsStartingWith("maven", oc)
+					exutil.DumpBuildLogs("openshift-jee-sample-docker", oc)
+					exutil.DumpDeploymentLogs("openshift-jee-sample", 1, oc)
+				}
+				debugAnyJenkinsFailure(br, oc.Namespace()+"-openshift-jee-sample", oc, true)
+				br.AssertSuccess()
 
-			g.By("verify job is in jenkins")
-			_, err = j.WaitForContent("", 200, 30*time.Second, "job/%s/job/%s-sample-pipeline/", oc.Namespace(), oc.Namespace())
-			o.Expect(err).NotTo(o.HaveOccurred())
+				// wait for the service to be running
+				g.By("expecting the openshift-jee-sample service to be deployed and running")
+				_, err = exutil.GetEndpointAddress(oc, "openshift-jee-sample")
+				o.Expect(err).NotTo(o.HaveOccurred())
 
-			g.By(fmt.Sprintf("delete pipeline strategy bc %q", origPipelinePath))
-			err = oc.Run("delete").Args("bc", "sample-pipeline").Execute()
-			o.Expect(err).NotTo(o.HaveOccurred())
+				g.By("clean up openshift resources for next potential run")
+				err = oc.Run("delete").Args("bc", "openshift-jee-sample").Execute()
```

Comment:
Could you use --selector to delete everything instead of having to delete each part (bc, is, svc, route) individually?  Also applies to a couple of other sections where you are deleting things maybe?
Example: `oc delete all --selector="label=cakephp-ex"`